### PR TITLE
Desktop icon

### PR DESCRIPTION
**Pull request by useredsa**
I made small changes to add Kakoune's icon to the desktop entry.
Current size is 124x124 which looks good in full-hd.

This is my first ever pull request. I hope I did everything in the good way!

### CODE REVIEW

**Comment by useredsa** [comment]
I changed the name of the entry because it was too long and it wouldn't fit in the space in Ubuntu. I set it to Kakoune (C). You can revert it to original state though.

**Comment by alexherbo2** [comment]
I prefer to not store images in the repository and use SVG.

https://github.com/mawww/kakoune/blob/master/doc/kakoune_logo.svg

**Comment by alexherbo2** [comment]
Can `$XDG_DATA_HOME/icons` be used instead of one of the `$XDG_DATA_DIRS`?

“Copy to `$XDG_DATA_DIRS/icons`” is confusing.  I would phrase it to one of the directories.

**Comment by useredsa** [comment]
> Can `$XDG_DATA_HOME/icons` be used instead of one of the `$XDG_DATA_DIRS`?
> 
> “Copy to `$XDG_DATA_DIRS/icons`” is confusing. I would phrase it to one of the directories.

According to [the docs](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html) it is possible but it is maintained for backwards compatibility. Additionally, it only serves for one user (which may be good). I think we can include both options.

About using svg, we can substitute `cp` by `curl` and remove the image from the repository. Do you want me to try this and modify the request?

**Comment by alexherbo2** [comment]
I do not see `$XDG_DATA_HOME/icons` in the spec.  Does it work?

You can keep the copy phrasing by giving a link to the logo.

> Copy the [Kakoune logo] to `$XDG_DATA_HOME/icons/kakoune.svg`.

[Kakoune logo]: https://github.com/mawww/kakoune/blob/master/doc/kakoune_logo.svg

I’m not sure how icons are found inside the `icons` directory.
Can it be directly a file, a directory with a certain structure?

https://wiki.archlinux.org/index.php/Desktop_entries#Application_entry

In the example, `Icon` value has no extension.

**Comment by useredsa** [comment]
Well, according to the [base dir specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), `$XDG_DATA_HOME` should be more important than `$XDG_DATA_DIRS` and used before. The same [link](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#referencing) states that if in any part of the specification they refer to `$XDG_DATA_DIRS`, a local user version may exist in `$XDG_DATA_HOME`. Therefore, yes, it's allowed to be there.

I tried it in one of my devices with Ubuntu and it worked putting it inside `$XDG_DATA_HOME/icons`, but it didn't when I put it in `$HOME/icons`, so I suppose the specification is wrong there. This [issue](https://gitlab.freedesktop.org/xdg/default-icon-theme/issues/12) argues it's wrong with the last location.

As this is only for one user, maybe we should include how to install for one user and how to install for everyone.

As for how icons are found, I must have done it wrong. It worked on my machine, and the telegram icon in my machine was also there, but this [part](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#install_icons) of the documentation clearly says that we should locate our icons in the default `hicolor` theme. I. e. inside `$prefix/share/icons/hicolor/scalable/apps` if we are going to store an `.svg`. I didn't have a folder for scalable icons in my machine, so I had to create one. It works fine with or without extension. I suppose in the later case, it can look for both `.png` and `.svg` files.

**Comment by alexherbo2** [comment]
I prefer to not mention `$XDG_DATA_DIRS`, but just `$XDG_DATA_HOME` – like I did for installing the desktop entry.

In the end, what is the full path to copy `kakoune.svg` and the `Icon` value?

**Comment by alexherbo2** [comment]
Do you mind creating an issue or a pull request on Kakoune for building and installing the icons?

So we can use them and it will not be specific to connect.kak.

**Comment by useredsa** [comment]
> In the end, what is the full path to copy kakoune.svg and the Icon value?

Updated everything in the new commit.

> Do you mind creating an issue or a pull request on Kakoune for building and installing the icons?
>
> So we can use them and it will not be specific to connect.kak.

I could ask to add a tutorial in the wiki about how to do it. However, in my case, I don't want both icons, because `connect` acts as a normal  kakoune session when it hasn't been linked with `:t` or `:T`. If you agree I'll write instructions in the forum and ask the admins to incorporate it to the wiki.

Finally, could you test everything works? I've deduced you use KDE. It'd be nice to have it tested in both GNOME and KDE.

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Add [`rc`] to your autoload or source [`connect.kak`] and its [modules] manually
 ### Add to desktop
 
 1. Copy [`kakoune-connect.desktop`] to `$XDG_DATA_HOME/applications`.
-2. Copy [`kakoune-icon.png`] to `$XDG_DATA_DIRS/icons`. Note that `$XDG_DATA_DIRS` is a list of folders.
+2. Copy [Kakoune's logo](https://github.com/mawww/kakoune/blob/master/doc/kakoune_logo.svg) to `$XDG_DATA_HOME/icons/hicolor/scalable/apps/kakoune.svg`.
```
Comment (alexherbo2) [inline README.md]: What is `hicolor`?

File: share/applications/kakoune-connect.desktop [hunk 2]
```diff
@@ -2,4 +2,4 @@
 Type=Application
 Name=Kakoune (C)
 Exec=kak-connect %F
-Icon=kakoune-icon.png
+Icon=kakoune.svg
```
Comment (alexherbo2) [inline share/applications/kakoune-connect.desktop:5]: Shouldn’t be `kakoune` without the extension?

File: README.md (hunk 1 above)
Comment (useredsa) [reply README.md]: > this part of the documentation clearly says that we should locate our icons in the default hicolor them.

It's the default color scheme. It is present on systems supporting xdg. Applications that don't change their icon according to the theme put their logos there.

File: share/applications/kakoune-connect.desktop (hunk 2 above)
Comment (useredsa) [reply share/applications/kakoune-connect.desktop:5]: The documentation doesn't suggest that. I think it is correct to put it with extension.

**Comment by useredsa** [comment]
Btw, I included this into my `kakrc`. It's pretty useful to keep things updated with the repository.

```
plug "alexherbo2/connect.kak" do %{
    ln -f "bin/kak-connect" "$HOME/bin/"
    ln -f "share/applications/kakoune-connect.desktop" "$XDG_DATA_HOME/applications/"
    curl "https://raw.githubusercontent.com/mawww/kakoune/master/doc/kakoune_logo.svg" --create-dirs -o "$XDG_DATA_HOME/icons/hicolor/scalable/apps/kakoune.svg"
}
```

You can also add
```
XDG_DATA_HOME="${XDG_DATA_HOME:-$HOME/.local/share}"
```
as a first sentence in case the variables aren't defined for `/bin/sh`.

File: share/applications/kakoune-connect.desktop (hunk 2 above)
Comment (useredsa) [reply share/applications/kakoune-connect.desktop:5]: Can we resolve this?

**Comment by alexherbo2** [comment]
Can you squash the commits?

**Comment by useredsa** [comment]
Done  :+1:

File: README.md [hunk 3]
```diff
@@ -18,8 +18,9 @@ Add [`rc`] to your autoload or source [`connect.kak`] and its [modules] manually
 ### Add to desktop
 
 1. Copy [`kakoune-connect.desktop`] to `$XDG_DATA_HOME/applications`.
-2. Add [`kak-connect`] to your path.
-3. Open `$XDG_CONFIG_HOME/mimeapps.list` and add the following MIME type association:
+2. Copy [Kakoune's logo](https://github.com/mawww/kakoune/blob/master/doc/kakoune_logo.svg) to `$XDG_DATA_HOME/icons/hicolor/scalable/apps/kakoune.svg`.
```
Comment (alexherbo2) [inline README.md]: Prefer reference to inline links.

Comment (useredsa) [reply README.md]: Done

File: README.md [hunk 4]
```diff
@@ -18,8 +18,9 @@ Add [`rc`] to your autoload or source [`connect.kak`] and its [modules] manually
 ### Add to desktop
 
 1. Copy [`kakoune-connect.desktop`] to `$XDG_DATA_HOME/applications`.
-2. Add [`kak-connect`] to your path.
-3. Open `$XDG_CONFIG_HOME/mimeapps.list` and add the following MIME type association:
+2. Copy Kakoune's logo (`https://github.com/mawww/kakoune/blob/master/doc/kakoune_logo.svg`) to `$XDG_DATA_HOME/icons/hicolor/scalable/apps/kakoune.svg`.
```
Comment (alexherbo2) [inline README.md]: It is not a reference link.

Comment (useredsa) [reply README.md]: Sorry, I don't know what you mean. You can write it yourself as you want it.

Comment (alexherbo2) [reply README.md]:
``` markdown
2. Copy [Kakoune’s logo] to `$XDG_DATA_HOME/icons/hicolor/scalable/apps/kakoune.svg`.

[Kakoune’s logo]: https://github.com/mawww/kakoune/blob/master/doc/kakoune_logo.svg
```

**Comment by useredsa** [comment]
I was thinking of developing a module to use with kitty and connect new windows (tiling windows inside kitty). Perhaps I'll write another pull request in some time.

**Pull request merged**